src/pkg/zoci/common.go

```diff
@@ -29,12 +31,15 @@ type Remote struct {
 
 // NewRemote returns an oras remote repository client and context for the given url
 // with zarf opination embedded
-func NewRemote(url string, platform ocispec.Platform, mods ...oci.Modifier) (*Remote, error) {
-	logger := slog.New(message.ZarfHandler{})
+func NewRemote(ctx context.Context, url string, platform ocispec.Platform, mods ...oci.Modifier) (*Remote, error) {
+	l := slog.New(message.ZarfHandler{})
+	if logger.Enabled(ctx) {
```

Comment:
After seeing this fn used a few times, this looks like an effective way to propagate feature flags through the system. Imo we should look to use this pattern in the future.